x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/FieldAttribute.java

```diff
@@ -127,7 +127,7 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeOptionalWriteable(parent);
             out.writeString(name());
             dataType().writeTo(out);
-            out.writeNamedWriteable(field);
+            ((PlanStreamOutput) out).writeEsField(field);
```

Comment:
I think I'd prefer the logic for this to be on `EsField` itself. So it's just `field.write(out)`.